Test adding ARI to a New Model Run (#245)
ccao-data/model-res-avm: analyses/ARI-model-test.qmd

  conn = con,
  "SELECT *
   FROM spatial.census
  where year = '2022' and geography = 'county'"
) %>%
  st_as_sf() %>%
  filter(name == "Cook")

tract_data <- working_data %>%
  distinct(loc_census_acs5_tract_geoid, .keep_all = TRUE) %>%
  inner_join(tract_geographies, by = c("loc_census_acs5_tract_geoid" = "geoid")) %>%
  st_as_sf()

tract_data_non_zero <- working_data_non_zero %>%
  distinct(loc_census_acs5_tract_geoid, .keep_all = TRUE) %>%
  inner_join(tract_geographies, by = c("loc_census_acs5_tract_geoid" = "geoid")) %>%
  st_as_sf()
```

# Context

The ARI is a tool intended to demonstrate the need to preserve affordability in areas that are becoming less affordable at a faster rate. Census tracts where affordability loss risk is greatest receive the highest scores in this index. The ARI uses American Community Survey data to measure change over time across various factors that indicate affordability loss:

- Median Household Income
- Median Home Value
- Families Below the Federal Poverty Level
- Housing Unit Vacancy
- Renter Tenancy
- Individuals Employed in Management, Business, Science and Arts Occupations
- Individuals with a Four-Year Degree (Bachelor’s) or Higher.

> ![](ARI Image.png)

Reviewer: **issue:** This image should either be part of the repo (bad) or loaded from its source website (better). Right now, it's not included in the PR so this report can't knit.

- 2 points - Family poverty rate meets first point threshold above and housing unit vacancy relative rate of change is below the State average relative rate of change or is below 3.5%
- 3 points - Family poverty rate meets first threshold (relative rate of change is below the State average relative rate of change); housing unit vacancy relative rate of change is below the State average relative rate of change or is below 3.5%; and 2 of the other factor thresholds are met
- 4 points - Family poverty rate is less than the State average and the relative rate of change is below the State average relative rate of change); housing unit vacancy relative rate of change is below the State average relative rate of change or is below 3.5%; and 3 of the other factor thresholds are met
- 5 points - Family poverty rate is less than the State average and the relative rate of change is below the State average relative rate of change); housing unit vacancy relative rate of change is below the State average relative rate of change or is below 3.5%; and 4 of the other factor thresholds are met

## Map of ARI by Census Tract, removing ARI of 0

> When removing ARIs of 0, the number of census tracts left in the database diminishes significantly. The number of pin10 observations goes from 1,117,715 to 596,243. My best assumption is that the missing tracts are not experiencing any risk of Affordability (not receiving 1 on any indicator).

Reviewer: ```suggestion
When removing ARIs of 0, the number of census tracts left in the data diminishes significantly. The number of PIN10 observations goes from 1,117,715 to 596,243. My best assumption is that the missing tracts are not experiencing any risk of Affordability (not receiving 1 on any indicator).
```

```{r, echo = FALSE}
tract_data_non_zero %>%
  ggplot() +
  geom_sf(aes(fill = ARI)) +
  geom_sf(data = cook, color = "black", fill = NA) +
  scale_fill_viridis_c(option = "viridis") +
  theme_void() +
  labs(
    fill = "ARI"
  ) +
  coord_sf(xlim = c(-88.4, -87.52398), ylim = c(41.5, 42.2))
```

> ## Map of ARI by Census Tract, Keeping ARI of 0

Reviewer: ```suggestion
## Map of ARI by Census Tract, Keeping ARI of 0

```

> When keeping the 0s in the map, you can see some clusters of affordability risk, most notably on the North-Western pipeline. With that being said, the map does not seem particularly conclusive regarding regional trends, where there are pockets of high ARIs in the far suburbs. Within the city limits, fives seem to be exclusively in the North, although there are a significant number of fours on the South Side. To be honest, that makes it out to be a pretty interesting metric, which may not align with others in our database.

Reviewer: ```suggestion
When keeping the 0s in the map, you can see some clusters of affordability risk, most notably on the North-Western pipeline and the far suburbs. That being said, the map does not seem particularly conclusive regarding regional trends. Within the city limits, ARI values of 5 seem to be exclusively in the North, although there are a significant number of 4s on the South Side. To be honest, that makes it out to be a pretty interesting metric, which may not align with others in our database.
```

```{r, echo = FALSE}
tract_data %>%
  ggplot() +
  geom_sf(aes(fill = ARI)) +
  geom_sf(data = cook, color = "black", fill = NA) +
  scale_fill_viridis_c(option = "viridis") +
  theme_void() +
  labs(
    fill = "ARI"
  ) +
  coord_sf(xlim = c(-88.4, -87.52398), ylim = c(41.5, 42.2))
```

## Does Median SHAP Value Differ by Neighborhood?

In general, it seems as though higher SHAP values relate to higher value neighborhoods. These are concentrated on the North side of the County, often in proximity to Lake Michigan. This is likely due to the model trying to find reasons to increase the value of homes in these areas. 

> ### Median Absolute SHAP Values by Tract

Reviewer: ```suggestion
### Median Absolute SHAP Values by Tract

```

```{r, echo = FALSE}
ggplot(data = tract_data) +
  geom_sf(aes(fill = shap_tract_mean)) +
  scale_fill_viridis_c(option = "viridis") +
  theme_void() +
  labs(
    fill = "Added $ Value"
  )
```

### Median Absolute SHAP Values by Tract, Removing Tracts with 0 ARI

> Some of the same patterns are visible when we look at the data without 0 ARI tracts. But, the map is generally more complicated to interpret. 

Reviewer: ```suggestion
Some of the same patterns are visible when we look at the data without 0 ARI tracts. But, the map is generally more complicated to interpret.

```

```{r, echo = FALSE}
ggplot(data = tract_data_non_zero) +
  geom_sf(aes(fill = shap_tract_mean)) +
  geom_sf(data = cook, color = "black", fill = NA) +
  scale_fill_viridis_c(option = "viridis") +
  theme_void() +
  labs(
    fill = "Added $ Value"
  ) +
  coord_sf(xlim = c(-88.4, -87.52398), ylim = c(41.5, 42.2))
```

### Median Absolute SHAP Values by Township, Keeping Tracts with 0 ARI

Looking at the data on a less granular level, we can see that the SHAP values are once again higher on the North side of the County. This reflects the findings on the tract level. 

```{r, echo = FALSE}
ggplot(data = tract_data) +
  geom_sf(aes(fill = shap_township_mean), color = NA) +
  scale_fill_viridis_c(option = "viridis") +
  theme_void() +
  labs(
    fill = "Added $ Value"
  )
```

> 

Reviewer: ```suggestion
```

## Median Absolute Values of the 90th percentile in Each Neighborhood

The impact of a feature can be hidden by looking at the median, rather than at the extremes. Because of this, we group by neighborhood, and select the 90th percentile of each. This map looks almost exactly like the previous map. With that being said, the scale for SHAP values doubled in $ value. 

```{r, echo = FALSE}
filtered_plot <- working_data %>%
  group_by(loc_census_acs5_tract_geoid) %>%
  mutate(threshold = quantile(abs(ARI_SHAP), 0.9)) %>%
  filter(abs(ARI_SHAP) >= threshold) %>%
  mutate(median_90th_percentile = median(ARI_SHAP)) %>%
  distinct(loc_census_acs5_tract_geoid, .keep_all = TRUE) %>%
  inner_join(tract_geographies, by = c("loc_census_acs5_tract_geoid" = "geoid")) %>%
  st_as_sf()

ggplot(data = filtered_plot) +
  geom_sf(aes(fill = median_90th_percentile)) +
  scale_fill_viridis_c(option = "viridis") +
  theme_void() +
  labs(
    fill = "Added $ Value"
  )
```

> 

Reviewer: ```suggestion
```

## Correlation between SHAP Value of ARI and ARI

One of the assumptions that can be attached to SHAPs is that if there is a large correlation between the feature value, and the SHAP value, then it is providing something meaningful in the model. In this case, the correlation is around than |.1|. This reinforces the existing perception that it does not add significant value. 

```{r, echo = FALSE}
x <- cor(working_data$ARI_SHAP, working_data$ARI, use = "complete.obs")
x <- round(x, 2)
print(paste("Correlation =", x))

x <- cor(working_data_non_zero$ARI_SHAP, working_data_non_zero$ARI, use = "complete.obs")
x <- round(x, 2)
print(paste("Correlation =", x, "when removing 0s"))
```

> ## Creating a map to see if there are spatial disparities in the inter-tract SHAP values

Reviewer: **nitpick:** Use consistent title case.
